ReactSkia/sdk/CurlNetworking.cpp

```diff
@@ -155,38 +161,85 @@ void CurlNetworking::processNetworkRequest(CURLM *curlMultiHandle) {
 
 }
 
-bool CurlNetworking::preparePostRequest(shared_ptr<CurlRequest> curlRequest, folly::dynamic data ) {
-  const char *dataPtr = NULL;
+size_t CurlNetworking::read_callback(void* ptr, size_t size, size_t nmemb, void* userdata) {
+
+  CurlRequest *curlRequest = (CurlRequest *)userdata;
+  size_t readSize = size * nmemb;
+
+  // Calculate the remaining data size to send
+  size_t remainingSize = curlRequest->uploadDataLength - curlRequest->uploadBufferOffset;
+
+  if(remainingSize == 0) { // no more data left to copy,stop the current transfer
+    return remainingSize;
+  }
+  // Determine the size to be copied
+  size_t copySize = (remainingSize < readSize) ? remainingSize : readSize;
+
+  // Copy the data into the buffer
+  memcpy(ptr, curlRequest->uploadDataPtr + curlRequest->uploadBufferOffset, copySize);
+
+  // Update the position for the next read
+  curlRequest->uploadBufferOffset += copySize;
+
+  return copySize;
+}
+
+
+bool CurlNetworking::prepareRequest(shared_ptr<CurlRequest> curlRequest, folly::dynamic data, string methodName ) {
   size_t dataSize = 0;
   bool status = false;
 
-  if(data["string"].c_str()) {
-    dataPtr = data["string"].c_str();
-    dataSize = data["string"].getString().length();
-  } else if(data["formData"].c_str()) {
-    RNS_LOG_NOT_IMPL;
-    return status;
-  } else if(data["blob"].c_str()) {
-    RNS_LOG_NOT_IMPL;
-    return status;
-  } else if(data["uri"].c_str()) {
-    RNS_LOG_NOT_IMPL;
-    return status;
-  } else if(data["base64"].c_str()) {
-    RNS_LOG_NOT_IMPL;
-    return status;
-  } else {
-    RNS_LOG_ERROR ("Unknown Data for Post Request\n") ;
-    return status;
+  if(strcmp(methodName.c_str(),"DELETE")) {
+    if(data["string"].c_str()) {
+      dataSize = data["string"].getString().length();
+      curlRequest->uploadDataPtr =(char *) malloc(dataSize);
+      strcpy(curlRequest->uploadDataPtr,data["string"].c_str());
+    } else if(data["formData"].c_str()) {
+      RNS_LOG_NOT_IMPL;
```

Comment:
Can you add not implemented msh in detail of what is  not implemented instead here , it don't give clear info on what is not implemented